spec/ruby/core/marshal/dump_spec.rb

```diff
@@ -458,6 +458,12 @@ def obj.foo; end
       obj = MarshalSpec::BasicObjectSubWithRespondToFalse.new
       Marshal.dump(obj).should == "\x04\bo:2MarshalSpec::BasicObjectSubWithRespondToFalse\x00"
     end
+
+    it "dumps without marshaling any attached finalizer" do
+      obj = Object.new
+      ObjectSpace.define_finalizer(obj, &:itself)
```

Comment:
FYI this causes:
```
1)
Marshal.dump with an Object dumps without marshaling any attached finalizer ERROR
ArgumentError: Can't create Binding from C level Proc
/home/eregon/code/rubyspec/core/marshal/dump_spec.rb:464:in `define_finalizer'
/home/eregon/code/rubyspec/core/marshal/dump_spec.rb:464:in `block (3 levels) in <top (required)>'
/home/eregon/code/rubyspec/core/marshal/dump_spec.rb:6:in `<top (required)>'
[| | ==================100%================== | 00:00:00]      0F      1E 
```
on CRuby 3.0+. Just on 2.7 it seems to work.
I modified the spec a bit to make it work on all CRuby versions: https://github.com/ruby/spec/commit/d44cd79a85cfd1293d08987692c5c0522b197ffa